### PR TITLE
Improve code and fix altrurist may revive unreported body sometime

### PR DESCRIPTION
**Pull request by hinhinarrrrrr**
Improve code and fix altrurist may revive unreported body  sometime

### CODE REVIEW

File: Roles/Crewmate/Altruist.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 using Hazel;
 using InnerNet;
 using TOHE.Roles.Core;
+using TOHE.Roles.Impostor;
```
Comment (NikoCat233) [inline Roles/Crewmate/Altruist.cs:5]: are these 2 usings still needed in code?

**Review by NikoCat233 — CHANGES_REQUESTED** [review]

File: Roles/Neutral/Vulture.cs [hunk 2]
```diff
@@ -115,9 +113,6 @@ public override void OnFixedUpdate(PlayerControl player, bool lowLoad, long nowT
     }
     public override bool OnCheckReportDeadBody(PlayerControl reporter, NetworkedPlayerInfo deadBody, PlayerControl killer)
     {
-        // Vulture was eat body
-        if (UnreportablePlayers.Contains(deadBody.PlayerId)) return false;
-
```
Comment (Tommy-XL) [inline Roles/Neutral/Vulture.cs:120]: Vulture will be eat one dead body multiple times

File: Roles/Impostor/Cleaner.cs [hunk 3]
```diff
@@ -44,12 +41,10 @@ public override void Add(byte playerId)
 
     public override bool OnCheckReportDeadBody(PlayerControl reporter, NetworkedPlayerInfo deadBody, PlayerControl killer)
     {
-        if (CleanerBodies.Contains(deadBody.PlayerId)) return false;
```
Comment (Tommy-XL) [inline Roles/Impostor/Cleaner.cs:47]: Also here

File: Roles/Crewmate/Coroner.cs [hunk 4]
```diff
@@ -97,7 +95,6 @@ public override bool OnTaskComplete(PlayerControl player, int completedTaskCount
     public override void SetAbilityButtonText(HudManager hud, byte playerId) => hud.ReportButton.OverrideText(GetString("CoronerReportButtonText"));
     public override bool OnCheckReportDeadBody(PlayerControl reporter, NetworkedPlayerInfo deadBody, PlayerControl killer)
     {
-        if (UnreportablePlayers.Contains(deadBody.PlayerId)) return false;
```
Comment (Tommy-XL) [inline Roles/Crewmate/Coroner.cs:100]: Same need add check

**Comment by Tommy-XL** [comment]
`This branch has conflicts that must be resolved`, lol

**Comment by hinhinarrrrrr** [comment]
> `This branch has conflicts that must be resolved`, lol

solved

**Pull request merged**